lib/ipmi_chassis.c

```diff
@@ -1714,9 +1714,9 @@ ipmi_chassis_main(struct ipmi_intf * intf, int argc, char ** argv)
 					/* data 1 */
 					{"valid", 0, (1<<7), (1<<7),
 						"Boot flags valid"},
-					{"persistent", 0, (1<<6), (1<<6),
+					{"persistent", 0, (0xe0), (1<<6),
 						"Changes are persistent for all future boots"},
-					{"efiboot", 0, (1<<5), (1<<5),
+					{"efiboot", 0, (0xe0), (1<<5),
```

Comment:
Sorry, this fix is plain wrong. The bug is not there. It is in line 1773 where it says `flags[op->i] &= op->mask` instead of `flags[op->i] &= ~(op->mask)`, thus wiping out all other flags instead of clearing the state of the flag being set. The table must contain 1s in mask for the bits that need to be cleared when a particular option is being set. Also, this code (the original one) needs to be cleaned of magic numbers.

Reply:
Yeah, I wasn't sure about that. I thought about the ``flags[op->i] &= ~(op->mask)`` but then I was scared I would break something else. So I went for the simplest "fix" :) Thanks!